### PR TITLE
Beauty component improvements. Two new fantasy component prefixes

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. Changed the component backend code slightly to allow a single component to hold the total score instead of spawning new components. This should fix cases, such as material stacks without the MATERIAL_NO_EFFECTS flag, where multiple `set_costum_materials` calls can be made and new beauty components spawned.

## Why It's Good For The Game
Code improvement and a feature.

## Changelog
:cl:
add: Added "beautiful" and "ugly" (plus synonyms) fantasy prefixes. They all affect the beauty of the item.
/:cl:

### CODE REVIEW

File: code/__DEFINES/misc.dm
Comment (Qustinnus) [inline code/__DEFINES/misc.dm:531]: I dont like wrappers like this that obfuscate everything, people should be not dumb enough to be able to do this themselves.

File: code/game/atoms.dm
Comment (Qustinnus) [inline code/game/atoms.dm:1486]: I don't like how much beauty code is suddenly in set_custom_materials, material behavior should be inside on removed and on apply. If this gets added it sets a precedent for doing things like this in set_custom_materials.

File: code/datums/materials/_material.dm
Comment (spookydonut) [inline code/datums/materials/_material.dm:133]: why are you doing this async?

File: code/game/objects/structures/statues.dm
Comment (spookydonut) [inline code/game/objects/structures/statues.dm:22]: same here

**Review by spookydonut — CHANGES_REQUESTED** [review]

File: code/datums/materials/_material.dm
Comment (Ghommie) [reply code/datums/materials/_material.dm:133]: A bandaid ninjanomnom suggested to floyd when he first added the component, to fix some unspecified issues they both forgot about now.

File: code/game/objects/structures/statues.dm
Comment (Ghommie) [reply code/game/objects/structures/statues.dm:22]: I'll remove the async and see how it goes.

**Comment by spookydonut** [comment]
@Ghommie dont do that, maintainers can restart travis if needed

**Comment by Ghommie** [comment]
Guess we'll have to wait until Jared's CI pr is merged anyway, seeing how Travis is busted (which I didn't fully realize until a hour ago or so).

**Comment by spookydonut** [comment]
@Ghommie unit tests are failing https://travis-ci.org/github/tgstation/tgstation/jobs/739564865

**Comment by spookydonut** [comment]
@Ghommie rebase/merge commit for working CI

**Pull request merged**
